### PR TITLE
added maxpool1d onnx operator

### PR DESCRIPTION
**Pull request by Arjun31415**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

#1714 

### Changes

_Summarize the problem being addressed and your solution._

### Testing

_Describe how these changes have been tested._

### CODE REVIEW

File: crates/burn-import/src/burn/node/max_pool1d.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                max_pool1d: MaxPool1d,
+                phantom: core::marker::PhantomData<B>,
+                device: burn::module::Ignored<B::Device>,
+            }
+
+            impl<B: Backend> Model <B> {
+                #[allow(unused_variables)]
+                pub fn new(device: &B::Device) -> Self {
+                    let max_pool1d = MaxPool1dConfig::new(3)
+                        .with_strides(1)
```
Comment (antimora) [inline crates/burn-import/src/burn/node/max_pool1d.rs:136]: `.with_strides(1)` should bd `.with_stride(1)` 

Currently the test is failing.

File: flake.lock
Comment (antimora) [inline flake.lock:1]: This file should be removed. 

You are welcome to add `flake.lock` into `.gitignore` so it does not get accidentally committed.

File: flake.nix
Comment (antimora) [inline flake.nix:1]: Same as above.

**Review by antimora — CHANGES_REQUESTED** [review]
Thank you for your help adding ONNX ops. Overall, it looks perfect with minor required changes. One unit test is failing and I commented on it. Here is the failed test:

image

File: flake.lock
Comment (Arjun31415) [reply flake.lock:1]: Apologies, I did not mean to commit it

**Comment by Arjun31415** [comment]
@antimora fixed. The flake being commited was an issue on my end ( I am using nixos and need to specify the dependencies required to build the project for reproducibility ).

**Review by antimora — APPROVED** [review]
LGTM

Thank you for your contribution!

**Pull request merged**